libs/optimizer/src/lib/fixtures/optimizer.fixtures.ts

```diff
@@ -0,0 +1,457 @@
+  nbSegments: number,
+  givenRules: ScoringRules,
+): OptimizerFixture {
+  const distance = (getPreciseDistance(from, intermediate) + getPreciseDistance(intermediate, to)) / 1000;
+  const multiplier = getFreeDistanceMultiplier(givenRules);
+  return {
+    givenRequest: {
+      track: concatTracks(
+        createSegments(
+          { ...from, alt: 0, timeSec: 0 },
+          { ...intermediate, alt: 0, timeSec: 60 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+        createSegments(
+          { ...intermediate, alt: 0, timeSec: 60 },
+          { ...to, alt: 0, timeSec: 120 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+      ),
+    },
+    givenRules,
+    expectedResult: {
+      score: distance * multiplier,
+      lengthKm: distance,
+      multiplier,
+      circuit: CircuitType.OpenDistance,
+      optimal: true,
+    },
+  };
+}
+
+/**
+ *
+ * @returns a fixture for a free distance track with two intermediate bypass points and it's expected score
+ * @param from LatLon of the starting point of the free distance
+ * @param intermediate1 LatLon of the first intermediate bypass point
+ * @param intermediate2 LatLon of the second intermediate bypass point
+ * @param to LatLon of the ending point of the free distance
+ * @param nbSegments number of segments to create between each given points
+ * @param givenRules the ScoringRules for computing the score
+ */
+export function createFreeDistance2PointsFixture(
+  from: LatLon,
+  intermediate1: LatLon,
+  intermediate2: LatLon,
+  to: LatLon,
+  nbSegments: number,
+  givenRules: ScoringRules,
+): OptimizerFixture {
+  const distance =
+    (getPreciseDistance(from, intermediate1) +
+      getPreciseDistance(intermediate1, intermediate2) +
+      getPreciseDistance(intermediate2, to)) /
+    1000;
+  const multiplier = getFreeDistanceMultiplier(givenRules);
+  return {
+    givenRequest: {
+      track: concatTracks(
+        createSegments(
+          { ...from, alt: 0, timeSec: 0 },
+          { ...intermediate1, alt: 0, timeSec: 60 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+        createSegments(
+          { ...intermediate1, alt: 0, timeSec: 60 },
+          { ...intermediate2, alt: 0, timeSec: 120 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+        createSegments(
+          { ...intermediate2, alt: 0, timeSec: 120 },
+          { ...to, alt: 0, timeSec: 180 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+      ),
+    },
+    givenRules,
+    expectedResult: {
+      score: distance * multiplier,
+      lengthKm: distance,
+      multiplier,
+      circuit: CircuitType.OpenDistance,
+      optimal: true,
+    },
+  };
+}
+
+/**
+ *
+ * @returns a fixture for a free distance track with three intermediate bypass points and it's expected score
+ * @param from LatLon of the starting point of the free distance
+ * @param intermediate1 LatLon of the first intermediate bypass point
```

Comment:
turnpoint?

Reply:
You're right! My bad :-/.
All comments changed